### PR TITLE
Rhmap 6003 appdata export

### PR DESCRIPTION
**Pull request by pb82**
# Motivation

Merge back all the changes from the appdata export feature branch into master.

ping @wei-lee @ziccardi

### CODE REVIEW

File: lib/cmd/fh3/appdata/export/status.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+/**
+ * Update the progressbar if possible. Don't use `bar#tick` as that will not
+ * set the current progress but add the new valud to the current progress.
+ */
+function printout(data, bar) {
+  if (bar && data.totalSteps && data.step >= 0) {
+    bar.update(data.step / data.totalSteps);
+  } else {
+    /* eslint disable-next-line no-console */
+    console.log('No progress information in response');
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata/export/status.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

File: lib/cmd/fh3/appdata/export/status.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      if (poller) {
+        clearInterval(poller);
+      }
+      return cb(null);
+    }
+
+    // Need to use `console#log` here until we introduce some other logging or outpur
+    // framework to fhc
+    /* eslint disable-next-line no-console */
+    console.log('Polling job progress. Press `q` to quit');
```
Comment (fh-qube) [inline lib/cmd/fh3/appdata/export/status.js]: MAJOR Remove this logging statement. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* The code changes conform to the contributing & code style guides for this component (if they exist)
* A component owner should review your change (we've tagged these above)
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* If updating server component this PR should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Ensure JIRA is up to date before & after merge (e.g. deployment notes, tester notes, release notes, fix version)
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.
Fhcap PR will be generated automatically.
To deploy your changes to a cluster place comment in this PR with pattern [x][y] where x=deploy and y=clustername you wish to deploy. 
You can also deploy your change manually using fhcap-cli

**Comment by wei-lee** [comment]
LGTM

**Pull request merged**